src/main/java/com/eternalcode/combat/fight/effect/FightEffectController.java

```diff
@@ -42,17 +43,20 @@ public void onTag(FightTagEvent event) {
             this.effectService.applyCustomEffect(player, key, value));
     }
 
-    @EventHandler
-    public void onQuit(PlayerQuitEvent event) {
+    @EventHandler(priority = EventPriority.MONITOR, ignoreCancelled = true)
+    public void onQuit(FightUntagEvent event) {
         if (!this.effectSettings.customEffectsEnabled) {
             return;
         }
 
-        Player player = event.getPlayer();
-        this.effectService.restoreActiveEffects(player);
+        if (event.getCause() == CauseOfUnTag.LOGOUT) {
+            Player player = Bukkit.getPlayer(event.getPlayer());
+
+            this.effectService.restoreActiveEffects(player);
+        }
```

Comment:
Combat effects should always be removed.
```suggestion
        Player player = event.getPlayer();
        this.effectService.restoreActiveEffects(player);
```

Reply:
I think it's my mistake, on enemy logout the plugin should remove stored effects, not restore them. Fixin it rn